### PR TITLE
PB-25843 Referenced Conditions

### PR DESCRIPTION
**Pull request by RafalOlszewskiOS**

### CODE REVIEW

**Comment by RafalOlszewskiOS** [comment]
@SeungChoi Please take a review, it's the same changes as in https://github.com/OneSpan/esl.sdk.net/pull/31 with fixes for your review comments

File: sdk/SDK.Tests/src/Internal/Conversion/ReferencedConditionsConverterTest.cs [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+
+        internal Boolean compareReferencedDocuments (Silanis.ESL.API.ReferencedDocument apiRefDoc, ReferencedDocument sdkRefDoc)
+        {
+            return apiRefDoc.DocumentId.Equals (sdkRefDoc.DocumentId) &&
+                compareReferencedFields (apiRefDoc.Fields, sdkRefDoc.Fields);
+        }
+
+        internal Boolean compareReferencedDocuments (List<Silanis.ESL.API.ReferencedDocument> apiRefDocs, List<ReferencedDocument> sdkRefDocs)
+        {
+            if (apiRefDocs.Count != sdkRefDocs.Count) {
```
Comment (SeungChoi) [inline sdk/SDK.Tests/src/Internal/Conversion/ReferencedConditionsConverterTest.cs]: Please move "{" to the next line.

File: sdk/SDK.Tests/src/Internal/Conversion/ReferencedConditionsConverterTest.cs [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+                return false;
+            }
+
+            for (int i = 0; i < apiRefDocs.Count; i++) {
```
Comment (SeungChoi) [inline sdk/SDK.Tests/src/Internal/Conversion/ReferencedConditionsConverterTest.cs]: Please move "{" to the next line.

File: sdk/SDK.Tests/src/Internal/Conversion/ReferencedConditionsConverterTest.cs [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+                if (!compareReferencedDocuments (apiRefDocs [i], sdkRefDocs [i])) {
```
Comment (SeungChoi) [inline sdk/SDK.Tests/src/Internal/Conversion/ReferencedConditionsConverterTest.cs]: Please move "{" to the next line.

File: sdk/Silanis.ESL.SDK/src/Internal/Conversion/ReferencedConditionsConverter.cs [hunk 4]
```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Silanis.ESL.SDK
+{
+    internal class ReferencedConditionsConverter
+    {
+        public static API.ReferencedConditions ToAPI(ReferencedConditions sdkReferencedConditions) 
+        {
+            if (sdkReferencedConditions == null) {
```
Comment (SeungChoi) [inline sdk/Silanis.ESL.SDK/src/Internal/Conversion/ReferencedConditionsConverter.cs]: Please move "{" to the next line.

File: sdk/Silanis.ESL.SDK/src/Internal/Conversion/ReferencedConditionsConverter.cs [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+            API.ReferencedConditions apiReferencedConditions = new API.ReferencedConditions();
+            apiReferencedConditions.PackageId = sdkReferencedConditions.PackageId;
+            apiReferencedConditions.Documents = apiReferencedDocuments;
+            return apiReferencedConditions;
+        }
+
+        public static ReferencedConditions ToSDK(API.ReferencedConditions apiReferencedConditions) 
+        {
+            if (apiReferencedConditions == null) {
```
Comment (SeungChoi) [inline sdk/Silanis.ESL.SDK/src/Internal/Conversion/ReferencedConditionsConverter.cs]: Please move "{" to the next line.

File: sdk/SDK.Tests/src/Internal/Conversion/ReferencedConditionsConverterTest.cs (hunk 1 above)
Comment (RafalOlszewskiOS) [reply sdk/SDK.Tests/src/Internal/Conversion/ReferencedConditionsConverterTest.cs]: Fixed

File: sdk/SDK.Tests/src/Internal/Conversion/ReferencedConditionsConverterTest.cs (hunk 2 above)
Comment (RafalOlszewskiOS) [reply sdk/SDK.Tests/src/Internal/Conversion/ReferencedConditionsConverterTest.cs]: Fixed.

File: sdk/SDK.Tests/src/Internal/Conversion/ReferencedConditionsConverterTest.cs (hunk 3 above)
Comment (RafalOlszewskiOS) [reply sdk/SDK.Tests/src/Internal/Conversion/ReferencedConditionsConverterTest.cs]: Fixed.

File: sdk/Silanis.ESL.SDK/src/Internal/Conversion/ReferencedConditionsConverter.cs (hunk 4 above)
Comment (RafalOlszewskiOS) [reply sdk/Silanis.ESL.SDK/src/Internal/Conversion/ReferencedConditionsConverter.cs]: Fixed.

File: sdk/Silanis.ESL.SDK/src/Internal/Conversion/ReferencedConditionsConverter.cs (hunk 5 above)
Comment (RafalOlszewskiOS) [reply sdk/Silanis.ESL.SDK/src/Internal/Conversion/ReferencedConditionsConverter.cs]: Fixed.

**Comment by RafalOlszewskiOS** [comment]
@SeungChoi I fixed all formating issues.

**Pull request merged**